Fix nix build

Mic92 opened the pull request:
I dropped the build of the old cli from the flake, where the default.nix already seemed to removed in a previous iterations. Then I updated flake.nix and codex-rs expression to be able to build again (see individual commits for details).

Tested by running the following builds:


```
$ nix build .#packages.x86_64-linux.codex-rs
$ nix build .#packages.aarch64-darwin.codex-cli
```

Mic92 committed:
flake: drop broken codex-cli
the default.nix for this was removed, so this package is no longer
functional:

% nix build .#packages.aarch64-darwin.codex-cli
error:
       … while calling the 'import' builtin
         at /nix/store/67mzwrmcffhiagq88zwnigb8npv6lxxv-source/flake.nix:28:21:
           27|         ];
           28|         codex-cli = import ./codex-cli {
             |                     ^
           29|           inherit pkgs monorepo-deps;

       error: opening file '/nix/store/67mzwrmcffhiagq88zwnigb8npv6lxxv-source/codex-cli/default.nix': No such file or directory

Signed-off-by: Jörg Thalheim <joerg@thalheim.io>

Mic92 committed:
flake: make codex-rs the new default package

also defaultPackage / defaultApps got deprecated by packages.default
quite a while ago, which is now dropped.

Signed-off-by: Jörg Thalheim <joerg@thalheim.io>

Mic92 committed:
flake: bump deps

the previous of nixpkgs was too old to build codex-rs

Signed-off-by: Jörg Thalheim <joerg@thalheim.io>

Mic92 committed:
nix/codex-rs: provide fod hash for ratatui

this is needed because ratatui fetched from git

Signed-off-by: Jörg Thalheim <joerg@thalheim.io>

Mic92 commented:
I have read the CLA Document and I hereby sign the CLA

0xadeeb reviewed (approved):
LGTM, but check if monorepo-deps in flakes.nix (pnpm and husky) is still required. If not you can remove them too

Mic92 committed:
flake: drop pnpm & husky

Mic92 committed:
flake: remove unused rust-overlay

it was configured but unused.

Mic92 committed:
flake: remove flake-utils

this flake is simple enough to not need it.

Mic92 committed:
flake: remove app output

this is just the same as packages in this case, nix run will work also
with packages outputs.

Mic92 committed:
flake: remove disfunctional devShell

The shell alias looks dubious and the shellHook also doesn't work:

bash: /nix/store/b010ibix7i333qhv5smcb129yly1dmfj-cargo-setup-hook.sh:
Is a directory

This devShell is actually not needed. nix develop will just fallback to
the package, which has already all dependencies.
on the upside we can now use just callPackage which also work better
with cross-compilation than the current approach.

Mic92 commented:
cc @bolinfest for review

pakrym-oai commented:
Thank you!
